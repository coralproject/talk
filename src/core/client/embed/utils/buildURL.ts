export default function buildURL({
  protocol = window.location.protocol,
  hostname = window.location.hostname,
  port = window.location.port,
  pathname = window.location.pathname,
  search = window.location.search,
  hash = window.location.hash,
} = {}) {
  if (search && search[0] !== "?") {
    search = `?${search}`;
  } else if (search === "?") {
    search = "";
  }
  return `${protocol}//${hostname}${
    port ? `:${port}` : ""
  }${pathname}${search}${hash}`;
}
