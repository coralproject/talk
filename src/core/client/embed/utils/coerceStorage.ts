import { createInMemoryStorage } from "coral-framework/lib/storage";

function coerceStorage(type: "localStorage" | "sessionStorage"): Storage {
  let storage: Storage;
  try {
    storage = window[type];
  } catch {
    // eslint-disable-next-line no-console
    console.debug(`${type} not available, use in-memory replacement`);

    storage = createInMemoryStorage();
  }

  return storage;
}

export default coerceStorage;
