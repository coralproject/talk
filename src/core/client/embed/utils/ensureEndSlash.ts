export default function ensureEndSlash(p: string) {
  return p.match(/\/$/) ? p : `${p}/`;
}
