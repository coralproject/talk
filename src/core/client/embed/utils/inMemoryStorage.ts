class InMemoryStorage implements Storage {
  private readonly data = new Map<string, string>();

  public get length() {
    return this.data.size;
  }

  public clear() {
    this.data.clear();
  }

  public key(n: number) {
    const keys = [...this.data.keys()];
    if (keys.length < n || n < 0) {
      return null;
    }

    return keys[n];
  }

  public getItem(key: string): string | null {
    return this.data.get(key) || null;
  }

  public removeItem(key: string): void {
    this.data.delete(key);
  }

  public setItem(key: string, value: string): void {
    this.data.set(key, value);
  }
}

export default InMemoryStorage;
