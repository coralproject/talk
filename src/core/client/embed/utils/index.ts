export { default as buildURL } from "./buildURL";
export { default as ensureEndSlash } from "./ensureEndSlash";
export { default as ensureNoEndSlash } from "./ensureNoEndSlash";
export { default as startsWith } from "./startsWith";
export { default as prefixStorage } from "./prefixStorage";
export { default as parseHashQuery } from "./parseHashQuery";
