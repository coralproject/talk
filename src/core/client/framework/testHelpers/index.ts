export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export { default as createSinonStub } from "./createSinonStub";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
export { default as createUUIDGenerator } from "./createUUIDGenerator";
export * from "./denormalize";
export { default as replaceHistoryLocation } from "./replaceHistoryLocation";
