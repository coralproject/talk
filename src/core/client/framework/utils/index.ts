export { default as buildURL } from "./buildURL";
export { default as parseURL } from "./parseURL";
export { default as modifyQuery } from "./modifyQuery";
