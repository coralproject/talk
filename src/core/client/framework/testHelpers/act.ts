import TestRenderer from "react-test-renderer";

import { isPromiseLike } from "coral-common/utils";

export default function act<T>(callback: () => T): T {
  let callbackResult: T;
  const actResult = TestRenderer.act(() => {
    callbackResult = callback();
    return callbackResult as any;
  });
  if (isPromiseLike(callbackResult!)) {
    // Return it this way, to preserve warnings that React emits.
    return {
      then(resolve: (value: any) => void, reject: (err?: Error) => void) {
        (actResult as PromiseLike<any>).then(() => {
          resolve(callbackResult);
        }, reject);
      },
    } as any;
  }
  return callbackResult!;
}
