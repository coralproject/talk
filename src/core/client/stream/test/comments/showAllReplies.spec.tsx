import { ReactTestRenderer } from "react-test-renderer";
import sinon from "sinon";

import { timeout } from "talk-common/utils";
import { createSinonStub } from "talk-framework/testHelpers";

import { assets, comments, settings } from "../fixtures";
import create from "./create";

let testRenderer: ReactTestRenderer;
beforeEach(() => {
  const commentStub = {
    ...comments[0],
    replies: createSinonStub(
      s => s.throws(),
      s =>
        s.withArgs({ first: 5, orderBy: "CREATED_AT_ASC" }).returns({
          edges: [
            {
              node: comments[1],
              cursor: comments[1].createdAt,
            },
          ],
          pageInfo: {
            endCursor: comments[1].createdAt,
            hasNextPage: true,
          },
        }),
      s =>
        s
          .withArgs({
            first: sinon.match(n => n > 10000),
            orderBy: "CREATED_AT_ASC",
            after: comments[1].createdAt,
          })
          .returns({
            edges: [
              {
                node: comments[2],
                cursor: comments[2].createdAt,
              },
            ],
            pageInfo: {
              endCursor: comments[2].createdAt,
              hasNextPage: false,
            },
          })
    ),
  };

  const assetStub = {
    ...assets[0],
    comments: {
      pageInfo: {
        hasNextPage: false,
      },
      edges: [
        {
          node: commentStub,
          cursor: commentStub.createdAt,
        },
      ],
    },
  };

  const resolvers = {
    Query: {
      settings: sinon.stub().returns(settings),
      comment: createSinonStub(
        s => s.throws(),
        s => s.withArgs(undefined, { id: commentStub.id }).returns(commentStub)
      ),
      asset: createSinonStub(
        s => s.throws(),
        s =>
          s
            .withArgs(undefined, { id: assetStub.id, url: null })
            .returns(assetStub)
      ),
    },
  };

  ({ testRenderer } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue(assetStub.id, "assetID");
    },
  }));
});

it("renders comment stream", async () => {
  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it("show all replies", async () => {
  testRenderer.root
    .findByProps({ id: `talk-comments-replyList-showAll--${comments[0].id}` })
    .props.onClick();

  // Wait for loading.
  await timeout();
  expect(testRenderer.toJSON()).toMatchSnapshot();
});
