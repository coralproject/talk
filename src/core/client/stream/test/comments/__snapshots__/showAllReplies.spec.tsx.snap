// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders comment stream 1`] = `
<div
  className="HorizontalGutter-root App-root HorizontalGutter-full"
>
  <ul
    className="TabBar-root TabBar-primary"
    role="tablist"
  >
    <li
      className="Tab-root"
      id="tab-COMMENTS"
      role="presentation"
    >
      <button
        aria-controls="tabPane-COMMENTS"
        aria-selected={true}
        className="BaseButton-root Tab-button Tab-primary Tab-active"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        role="tab"
        type="button"
      >
        <span>
          ⁨2⁩ Comments
        </span>
      </button>
    </li>
  </ul>
  <section
    aria-labelledby="tab-COMMENTS"
    className="App-tabContent"
    id="tabPane-COMMENTS"
    role="tabpanel"
  >
    <div
      className="HorizontalGutter-root Stream-root HorizontalGutter-double"
    >
      <div
        className="HorizontalGutter-root HorizontalGutter-half"
      >
        <div
          className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter"
        >
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign in
          </button>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantOutlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Register
          </button>
        </div>
        <div
          className="HorizontalGutter-root PostCommentFormFake-root HorizontalGutter-full"
        >
          <div
            aria-hidden="true"
          >
            <div>
              <div
                className=""
              >
                <div
                  aria-hidden="true"
                  className="RTE-placeholder RTE-placeholder "
                >
                  Post a comment
                </div>
                <div
                  aria-placeholder="Post a comment"
                  className="RTE-contentEditable RTE-content  RTE-contentEditableDisabled"
                  contentEditable={false}
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                  disabled={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onCut={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyDown={[Function]}
                  onPaste={[Function]}
                  onSelect={[Function]}
                />
                <div
                  className="RTE-toolbar  RTE-toolbarDisabled RTE-toolbarBottom Toolbar-toolbar"
                >
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Bold"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_bold
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Italic"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_italic
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Blockquote"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_quote
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <button
            className="BaseButton-root Button-root Button-sizeRegular Button-colorPrimary Button-variantFilled Button-fullWidth Button-disabled"
            disabled={true}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="submit"
          >
            Sign in and Join the Conversation
          </button>
        </div>
      </div>
      <div
        aria-live="polite"
        className="HorizontalGutter-root HorizontalGutter-full"
        id="talk-comments-stream-log"
        role="log"
      >
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="Indent-root"
            >
              <div
                className=""
              >
                <div
                  className="Comment-root"
                  role="article"
                >
                  <div
                    className="Flex-root Flex-flex Flex-justifySpaceBetween Flex-directionRow"
                    id="comment-comment-0"
                  >
                    <div
                      className="Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
                    >
                      <span
                        className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                      >
                        Markus
                      </span>
                      <div
                        className="Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow"
                      >
                        <time
                          className="Timestamp-root RelativeTime-root"
                          dateTime="2018-07-06T18:24:00.000Z"
                          title="2018-07-06T18:24:00.000Z"
                        >
                          2018-07-06T18:24:00.000Z
                        </time>
                      </div>
                    </div>
                  </div>
                  <div
                    className="HorizontalGutter-root HorizontalGutter-full"
                  >
                    <div
                      className="HTMLContent-root"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Joining Too",
                        }
                      }
                    />
                    <div
                      className="Flex-root Flex-flex Flex-halfItemGutter Flex-directionRow"
                    >
                      <button
                        className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                        id="comments-commentContainer-replyButton-comment-0"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        type="button"
                      >
                        <span>
                          Reply
                        </span>
                      </button>
                      <div
                        className="Popover-root"
                      >
                        <button
                          aria-controls="permalink-popover-comment-0"
                          className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          type="button"
                        >
                          <span>
                            Share
                          </span>
                        </button>
                        <div
                          aria-hidden={true}
                          aria-labelledby="permalink-popover-comment-0-ariainfo"
                          id="permalink-popover-comment-0"
                          role="popup"
                        >
                          <div
                            className="AriaInfo-root"
                            id="permalink-popover-comment-0-ariainfo"
                          >
                            A dialog showing a permalink to the comment
                          </div>
                        </div>
                      </div>
                      <button
                        className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        type="button"
                      >
                        <span>
                          Respect
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
            id="talk-comments-replyList-log--comment-0"
            role="log"
          >
            <div
              className="HorizontalGutter-root HorizontalGutter-full"
            >
              <div
                className="HorizontalGutter-root HorizontalGutter-full"
              >
                <div
                  className="Indent-root"
                >
                  <div
                    className="Indent-level1"
                  >
                    <div
                      className="Comment-root"
                      role="article"
                    >
                      <div
                        className="Flex-root Flex-flex Flex-justifySpaceBetween Flex-directionRow"
                        id="comment-comment-1"
                      >
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
                        >
                          <span
                            className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                          >
                            Lukas
                          </span>
                          <div
                            className="Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow"
                          >
                            <time
                              className="Timestamp-root RelativeTime-root"
                              dateTime="2018-07-06T18:24:00.000Z"
                              title="2018-07-06T18:24:00.000Z"
                            >
                              2018-07-06T18:24:00.000Z
                            </time>
                          </div>
                        </div>
                      </div>
                      <div
                        className="HorizontalGutter-root HorizontalGutter-full"
                      >
                        <div
                          className="HTMLContent-root"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "What's up?",
                            }
                          }
                        />
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-directionRow"
                        >
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            id="comments-commentContainer-replyButton-comment-1"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Reply
                            </span>
                          </button>
                          <div
                            className="Popover-root"
                          >
                            <button
                              aria-controls="permalink-popover-comment-1"
                              className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                              onTouchEnd={[Function]}
                              type="button"
                            >
                              <span>
                                Share
                              </span>
                            </button>
                            <div
                              aria-hidden={true}
                              aria-labelledby="permalink-popover-comment-1-ariainfo"
                              id="permalink-popover-comment-1"
                              role="popup"
                            >
                              <div
                                className="AriaInfo-root"
                                id="permalink-popover-comment-1-ariainfo"
                              >
                                A dialog showing a permalink to the comment
                              </div>
                            </div>
                          </div>
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Respect
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="Indent-root"
            >
              <div
                className="Indent-level1 Indent-noBorder"
              >
                <button
                  aria-controls="talk-comments-replyList-log--comment-0"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular Button-variantOutlined Button-fullWidth"
                  disabled={false}
                  id="talk-comments-replyList-showAll--comment-0"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  Show All
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`show all replies 1`] = `
<div
  className="HorizontalGutter-root App-root HorizontalGutter-full"
>
  <ul
    className="TabBar-root TabBar-primary"
    role="tablist"
  >
    <li
      className="Tab-root"
      id="tab-COMMENTS"
      role="presentation"
    >
      <button
        aria-controls="tabPane-COMMENTS"
        aria-selected={true}
        className="BaseButton-root Tab-button Tab-primary Tab-active"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        role="tab"
        type="button"
      >
        <span>
          ⁨2⁩ Comments
        </span>
      </button>
    </li>
  </ul>
  <section
    aria-labelledby="tab-COMMENTS"
    className="App-tabContent"
    id="tabPane-COMMENTS"
    role="tabpanel"
  >
    <div
      className="HorizontalGutter-root Stream-root HorizontalGutter-double"
    >
      <div
        className="HorizontalGutter-root HorizontalGutter-half"
      >
        <div
          className="Flex-root Flex-flex Flex-itemGutter Flex-alignCenter"
        >
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantUnderlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Sign in
          </button>
          <button
            className="BaseButton-root Button-root Button-sizeSmall Button-colorPrimary Button-variantOutlined"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="button"
          >
            Register
          </button>
        </div>
        <div
          className="HorizontalGutter-root PostCommentFormFake-root HorizontalGutter-full"
        >
          <div
            aria-hidden="true"
          >
            <div>
              <div
                className=""
              >
                <div
                  aria-hidden="true"
                  className="RTE-placeholder RTE-placeholder "
                >
                  Post a comment
                </div>
                <div
                  aria-placeholder="Post a comment"
                  className="RTE-contentEditable RTE-content  RTE-contentEditableDisabled"
                  contentEditable={false}
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                  disabled={true}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onCut={[Function]}
                  onFocus={[Function]}
                  onInput={[Function]}
                  onKeyDown={[Function]}
                  onPaste={[Function]}
                  onSelect={[Function]}
                />
                <div
                  className="RTE-toolbar  RTE-toolbarDisabled RTE-toolbarBottom Toolbar-toolbar"
                >
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Bold"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_bold
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Italic"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_italic
                    </span>
                  </button>
                  <button
                    className="Button-button"
                    disabled={false}
                    onClick={[Function]}
                    title="Blockquote"
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                      className="Icon-root Icon-md"
                    >
                      format_quote
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <button
            className="BaseButton-root Button-root Button-sizeRegular Button-colorPrimary Button-variantFilled Button-fullWidth Button-disabled"
            disabled={true}
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            onTouchEnd={[Function]}
            type="submit"
          >
            Sign in and Join the Conversation
          </button>
        </div>
      </div>
      <div
        aria-live="polite"
        className="HorizontalGutter-root HorizontalGutter-full"
        id="talk-comments-stream-log"
        role="log"
      >
        <div
          className="HorizontalGutter-root HorizontalGutter-full"
        >
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
          >
            <div
              className="Indent-root"
            >
              <div
                className=""
              >
                <div
                  className="Comment-root"
                  role="article"
                >
                  <div
                    className="Flex-root Flex-flex Flex-justifySpaceBetween Flex-directionRow"
                    id="comment-comment-0"
                  >
                    <div
                      className="Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
                    >
                      <span
                        className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                      >
                        Markus
                      </span>
                      <div
                        className="Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow"
                      >
                        <time
                          className="Timestamp-root RelativeTime-root"
                          dateTime="2018-07-06T18:24:00.000Z"
                          title="2018-07-06T18:24:00.000Z"
                        >
                          2018-07-06T18:24:00.000Z
                        </time>
                      </div>
                    </div>
                  </div>
                  <div
                    className="HorizontalGutter-root HorizontalGutter-full"
                  >
                    <div
                      className="HTMLContent-root"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "Joining Too",
                        }
                      }
                    />
                    <div
                      className="Flex-root Flex-flex Flex-halfItemGutter Flex-directionRow"
                    >
                      <button
                        className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                        id="comments-commentContainer-replyButton-comment-0"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        type="button"
                      >
                        <span>
                          Reply
                        </span>
                      </button>
                      <div
                        className="Popover-root"
                      >
                        <button
                          aria-controls="permalink-popover-comment-0"
                          className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          type="button"
                        >
                          <span>
                            Share
                          </span>
                        </button>
                        <div
                          aria-hidden={true}
                          aria-labelledby="permalink-popover-comment-0-ariainfo"
                          id="permalink-popover-comment-0"
                          role="popup"
                        >
                          <div
                            className="AriaInfo-root"
                            id="permalink-popover-comment-0-ariainfo"
                          >
                            A dialog showing a permalink to the comment
                          </div>
                        </div>
                      </div>
                      <button
                        className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseOut={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        type="button"
                      >
                        <span>
                          Respect
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="HorizontalGutter-root HorizontalGutter-full"
            id="talk-comments-replyList-log--comment-0"
            role="log"
          >
            <div
              className="HorizontalGutter-root HorizontalGutter-full"
            >
              <div
                className="HorizontalGutter-root HorizontalGutter-full"
              >
                <div
                  className="Indent-root"
                >
                  <div
                    className="Indent-level1"
                  >
                    <div
                      className="Comment-root"
                      role="article"
                    >
                      <div
                        className="Flex-root Flex-flex Flex-justifySpaceBetween Flex-directionRow"
                        id="comment-comment-1"
                      >
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
                        >
                          <span
                            className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                          >
                            Lukas
                          </span>
                          <div
                            className="Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow"
                          >
                            <time
                              className="Timestamp-root RelativeTime-root"
                              dateTime="2018-07-06T18:24:00.000Z"
                              title="2018-07-06T18:24:00.000Z"
                            >
                              2018-07-06T18:24:00.000Z
                            </time>
                          </div>
                        </div>
                      </div>
                      <div
                        className="HorizontalGutter-root HorizontalGutter-full"
                      >
                        <div
                          className="HTMLContent-root"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "What's up?",
                            }
                          }
                        />
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-directionRow"
                        >
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            id="comments-commentContainer-replyButton-comment-1"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Reply
                            </span>
                          </button>
                          <div
                            className="Popover-root"
                          >
                            <button
                              aria-controls="permalink-popover-comment-1"
                              className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                              onTouchEnd={[Function]}
                              type="button"
                            >
                              <span>
                                Share
                              </span>
                            </button>
                            <div
                              aria-hidden={true}
                              aria-labelledby="permalink-popover-comment-1-ariainfo"
                              id="permalink-popover-comment-1"
                              role="popup"
                            >
                              <div
                                className="AriaInfo-root"
                                id="permalink-popover-comment-1-ariainfo"
                              >
                                A dialog showing a permalink to the comment
                              </div>
                            </div>
                          </div>
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Respect
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="HorizontalGutter-root HorizontalGutter-full"
            >
              <div
                className="HorizontalGutter-root HorizontalGutter-full"
              >
                <div
                  className="Indent-root"
                >
                  <div
                    className="Indent-level1"
                  >
                    <div
                      className="Comment-root"
                      role="article"
                    >
                      <div
                        className="Flex-root Flex-flex Flex-justifySpaceBetween Flex-directionRow"
                        id="comment-comment-2"
                      >
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn"
                        >
                          <span
                            className="Typography-root Typography-heading3 Typography-colorTextPrimary Username-root"
                          >
                            Isabelle
                          </span>
                          <div
                            className="Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow"
                          >
                            <time
                              className="Timestamp-root RelativeTime-root"
                              dateTime="2018-07-06T18:24:00.000Z"
                              title="2018-07-06T18:24:00.000Z"
                            >
                              2018-07-06T18:24:00.000Z
                            </time>
                          </div>
                        </div>
                      </div>
                      <div
                        className="HorizontalGutter-root HorizontalGutter-full"
                      >
                        <div
                          className="HTMLContent-root"
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "Hey!",
                            }
                          }
                        />
                        <div
                          className="Flex-root Flex-flex Flex-halfItemGutter Flex-directionRow"
                        >
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            id="comments-commentContainer-replyButton-comment-2"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Reply
                            </span>
                          </button>
                          <div
                            className="Popover-root"
                          >
                            <button
                              aria-controls="permalink-popover-comment-2"
                              className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onMouseOut={[Function]}
                              onMouseOver={[Function]}
                              onTouchEnd={[Function]}
                              type="button"
                            >
                              <span>
                                Share
                              </span>
                            </button>
                            <div
                              aria-hidden={true}
                              aria-labelledby="permalink-popover-comment-2-ariainfo"
                              id="permalink-popover-comment-2"
                              role="popup"
                            >
                              <div
                                className="AriaInfo-root"
                                id="permalink-popover-comment-2-ariainfo"
                              >
                                A dialog showing a permalink to the comment
                              </div>
                            </div>
                          </div>
                          <button
                            className="BaseButton-root Button-root Button-sizeSmall Button-colorRegular Button-variantGhost"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseOut={[Function]}
                            onMouseOver={[Function]}
                            onTouchEnd={[Function]}
                            type="button"
                          >
                            <span>
                              Respect
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
